### PR TITLE
Patch/selfassign

### PR DESCRIPTION
**Pull request by binaryoverload**
- [x] Completed PR - Has all features or fixes that it was meant to and isn't half done
- [x] Follows the style guide - Style guide found [here](https://github.com/FlareBot/FlareBot/blob/master/CONTRIBUTING.md)
- [x] Your code follows the same code style as the rest of the project
- [x] Tested changes

## What is fixed
Self Assign throwing errors when encountering deleted roles

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -37,7 +37,12 @@
         <dependency>
             <groupId>net.dv8tion</groupId>
             <artifactId>JDA</artifactId>
-            <version>3.1.0_206</version>
+            <version>3.1.1_213</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-lang3</artifactId>
```
Comment (ArsenArsen) [inline pom.xml]: In JDA already

File: src/main/java/stream/flarebot/flarebot/Events.java [hunk 2]
```diff
@@ -367,6 +368,13 @@ public void onDisconnect(DisconnectEvent event) {
                     .getClientCloseFrame().getCloseReason()));
     }
 
+    @Override
+    public void onRoleDelete(RoleDeleteEvent event) {
+        if (FlareBotManager.getInstance().getSelfAssignRoles(event.getGuild().getId()).contains(event.getRole().getId())) {
```
Comment (ArsenArsen) [inline src/main/java/stream/flarebot/flarebot/Events.java]: And now you inline variables...

File: src/main/java/stream/flarebot/flarebot/music/extractors/YouTubeExtractor.java [hunk 3]
```diff
@@ -32,18 +36,20 @@
...
             MessageUtils.editMessage(MessageUtils.getEmbed(user)
-                    .setDescription("Could not get that video/playlist!")
-                    .addField("YouTube said: ", e.getMessage(), true), message);
+                    .setDescription("Could not get that video/playlist! Make sure the URL is correct!")
+                    .setColor(Color.RED), message);
+            return;
+        } catch (YoutubeAccessException e) {
+            MessageUtils.editMessage(MessageUtils.getEmbed(user)
+                    .setDescription("Youtube could not be reached! Try again in a few minutes!\n" +
+                            "If the error continues, join our support discord: https://discord.gg/TTAUGvZ")
```
Comment (ArsenArsen) [inline src/main/java/stream/flarebot/flarebot/music/extractors/YouTubeExtractor.java]: Why would it be our job to tailor youtube being down?

File: src/main/java/stream/flarebot/flarebot/util/ErrorCatcher.java [hunk 4]
```diff
@@ -25,6 +25,9 @@ public FilterReply decide(ILoggingEvent event) {
         String msg = event.getFormattedMessage();
         if (msg == null)
             msg = "null";
+        if (event.getThreadName().startsWith("lava-daemon-pool")){
```
Comment (ArsenArsen) [inline src/main/java/stream/flarebot/flarebot/util/ErrorCatcher.java]: Whats the point?

File: src/main/java/stream/flarebot/flarebot/util/GeneralUtils.java [hunk 5]
```diff
@@ -72,4 +82,24 @@ public static String formatCommandPrefix(TextChannel channel, String usage) {
...
         return usage.replaceAll("\\{%\\}", prefix);
     }
+
+    public static AudioItem resolveItem(Player player, String input) throws IllegalArgumentException, YoutubeAccessException {
+        Optional<AudioItem> item = Optional.empty();
+        boolean failed = false;
+        for (int i = 0; i <= 2; i++) {
+            try {
+                item = Optional.ofNullable(player.resolve(input));
+                failed = false;
```
Comment (ArsenArsen) [inline src/main/java/stream/flarebot/flarebot/util/GeneralUtils.java]: You don't do a backoff here but spam youtube instead. Also break the loop on success

File: src/main/java/stream/flarebot/flarebot/util/GeneralUtils.java [hunk 6]
```diff
@@ -72,4 +82,24 @@ public static String formatCommandPrefix(TextChannel channel, String usage) {
...
+            } catch (FriendlyException | InterruptedException | ExecutionException e) {
```
Comment (ArsenArsen) [inline src/main/java/stream/flarebot/flarebot/util/GeneralUtils.java]: `Exception e`, and set the cause

**Comment by binaryoverload** [comment]
Fuck it

File: src/main/java/stream/flarebot/flarebot/util/GeneralUtils.java (hunk 6 above)
Comment (binaryoverload) [inline src/main/java/stream/flarebot/flarebot/util/GeneralUtils.java]: Catching `Exception` is a bad habit and should be avoided

**Pull request closed without merging**